image/manifest.go

```diff
@@ -89,7 +89,7 @@ func (m *manifest) validate(w walker) error {
 
 func (m *manifest) unpack(w walker, dest string) error {
 	for _, d := range m.Layers {
-		if d.MediaType != string(schema.MediaTypeImageConfig) {
+		if d.MediaType != string(schema.MediaTypeImageLayer) {
 			continue
```

Comment:
On Fri, Sep 09, 2016 at 03:07:52AM -0700, Antonio Murdaca wrote:
> > @@ -89,7 +89,7 @@ func (m *manifest) validate(w walker) error {
> >  
> >  func (m *manifest) unpack(w walker, dest string) error {
> >  	for _, d := range m.Layers {
> > -		if d.MediaType != string(schema.MediaTypeImageConfig) {
> > +		if d.MediaType != string(schema.MediaTypeImageLayer) {
> >  			continue
> >
> > Although the spec does not currently say that layers MUST have a
> > particular type, so this is currently an:
> > 
> > unimplemented: unpacking layer with media-type %q
> >
> > I expect the spec should clarify if it places bounds (upper or
> > lower) on manifest layer[] media types, and once that happens we
> > can adjust this to match.
> 
> is there an issue for this?

There are ~50 million open validation issues at the moment, so I
haven't opened up one specifically for this ;).  I try and bring it up
where it impacts an existing issue/PR.  Most recently in #286.